### PR TITLE
Implement structured logging

### PR DESCRIPTION
**Pull request by jkroepke**
HI,

thanks for you work! This PR implements structured logging. In kubernetes, the json log format can be used and passed to a ELK stack.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,13 @@
-FROM golang:1.18
+FROM golang:1.20
 
 COPY . /work
 WORKDIR /work
-RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o prometheus-nginxlog-exporter
+RUN CGO_ENABLED=0 GOOS=linux go build -ldflags '-w -s' -a -installsuffix cgo -o prometheus-nginxlog-exporter
```
Comment (jkroepke) [inline Dockerfile:5]: the ldflags removing debug symbols from the binary.

File: log/main.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package log
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type Logger struct {
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply log/main.go:8]: This Logger Interface is required for the tail library. The logger interface of the tail library requires Print functions which are not present in zap by default.

**Comment by jkroepke** [comment]
@martin-helmich what did you thing about this, after I would rebase this?

**Comment by jkroepke** [comment]
rebased

**Review by martin-helmich — APPROVED** [review]
I think I broke this when merging #333; I've resolved the conflicts and all's looking well, so I'll go ahead and merge this one as well. Thanks for your contributions!

**Pull request merged**